anchore_engine/services/policy_engine/engine/feeds/feeds.py

```diff
@@ -63,99 +67,146 @@
 from anchore_engine.utils import rfc3339str_to_datetime
 
 
-def build_group_sync_result(group=None, status="failure"):
-    return {
-        "group": group,
-        "status": status,
-        "total_time_seconds": 0,
-        "updated_record_count": 0,
-        "updated_image_count": 0,
-    }
+@dataclass
+class GroupSyncResult:
+    group: Optional[str] = None
+    status: str = "failure"
+    total_time_seconds: int = 0
+    updated_record_count: int = 0
+    updated_image_count: int = 0
 
 
-def build_feed_sync_results(feed=None, status="failure"):
-    return {"feed": feed, "status": status, "total_time_seconds": 0, "groups": []}
+@dataclass
+class FeedSyncResult:
+    feed: Optional[str] = None
+    status: str = "failure"
+    total_time_seconds: int = 0
+    groups: List[GroupSyncResult] = field(default_factory=list)
 
 
-class DataFeed(object):
+class LogContext:
```

Comment:
nit: I think this is good intermediate step, but we should eventually use a logger for this module that has this context embedded (e.g. a task can initialize its logger with the op id embedded or in the thread context). Good start, and we can continue this kind of pattern across the codebase.